crates/rpc/src/v03/method/simulate_transaction.rs

```diff
@@ -0,0 +1,665 @@
+        pub constructor_invocation: Option<FunctionInvocation>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub fee_transfer_invocation: Option<FunctionInvocation>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub validate_invocation: Option<FunctionInvocation>,
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct InvokeTxnTrace {
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub execute_invocation: Option<FunctionInvocation>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub fee_transfer_invocation: Option<FunctionInvocation>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub validate_invocation: Option<FunctionInvocation>,
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct L1HandlerTxnTrace {
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub function_invocation: Option<FunctionInvocation>,
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct SimulatedTransaction {
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub fee_estimation: Option<FeeEstimate>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub transaction_trace: Option<TransactionTrace>,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use pathfinder_common::{felt, Chain, TransactionVersion};
+    use pathfinder_storage::{JournalMode, Storage};
+    use starknet_gateway_test_fixtures::zstd_compressed_contracts::{
+        DUMMY_ACCOUNT, DUMMY_ACCOUNT_CLASS_HASH,
+    };
+
+    use crate::v02::types::reply::FeeEstimate;
+
+    use super::*;
+
+    mod tempdir {
+        use std::path::PathBuf;
+
+        pub(crate) fn new(path: PathBuf) -> TempDir {
+            if !path.exists() {
+                let _ = std::fs::create_dir_all(&path);
+            }
+            TempDir::new(path)
+        }
+
+        pub(crate) struct TempDir {
+            path: PathBuf,
+        }
+
+        impl TempDir {
+            pub(crate) fn new(path: PathBuf) -> Self {
+                Self { path }
+            }
+
+            pub(crate) fn file(&self, name: &str) -> PathBuf {
+                let mut buf = self.path.to_path_buf();
+                buf.push(name);
+                buf
+            }
+        }
+
+        impl Drop for TempDir {
+            fn drop(&mut self) {
+                if self.path.exists() && self.path.is_dir() {
+                    let _ = std::fs::remove_dir_all(&self.path);
+                }
+            }
+        }
+    }
+
+    #[tokio::test]
+    async fn test_simulate_transaction_deploy_balance_contract() {
```

Comment:
I don't know, you insisted on adding it.

Reply:
No,  I insisted on _not_ adding yet another class definition fixture for no good reason.

Reply:
Done, DUMMY_ACCOUNT test is removed.

Reply:
Hmm, and how does that help with _not_ adding yet another class definition fixture?

I think we should keep the dummy account one and remove this one.

Reply:
I am confused :) So you copy-pasted my perfectly fine and working test, switched it to existing contract, and declare this "new" test is valuable but the original one pointless, then demand that original test is removed?

Reply:
Not at all, I've _never_ written your test is pointless. I was suggesting that instead of introducing a new class used by the test we could modify that test to use a class fixture we already have. I see no downside to doing that and don't really understand _why_ you think we shouldn't do that.

Reply:
Getting a little too heated here..

@sergey-melnychuk I think the point was that if we can just re-use existing fixtures, then that is preferred. No point in adding more fixtures -- unless there is reasonable motivation of course.

We already have many fixtures that are difficult to track and know if they are still actually required or useful or, more crucial, still applicable (could be obsoltete due to changes on the gateway for example). So less is better in that sense.

So for example, if you're saying the fixture is good to have because it lets us verify against the gateway's output right now, whereas the existing ones don't - then sure. Or if it exhibits better traces, or exercises more stuff.

If you still feel strongly that this should be in, then lets leave it in. I would suggest having it follow the same format as the other fixtures (`.json.ztsd`) so that it can be used elsewhere as well. And maybe that the python test should also use the same one so they're comparable?

Reply:
Sorry, I feel like I increased the conversation degree for no good reason. Using existing contracts makes not much difference, but bloating the repo with binary data (dead weight) can indeed make check-out/CI jobs slower.